qiskit/transpiler/passes/routing/basic_swap.py

```diff
@@ -72,8 +72,11 @@ def run(self, dag):
 
         if len(dag.qubits) > len(self.coupling_map.physical_qubits):
             raise TranspilerError("The layout does not match the amount of qubits in the DAG")
-        disjoint_utils.require_layout_isolated_to_component(dag, self.coupling_map)
-
+        if self.target is not None:
+            target = self.target
+        else:
+            target = self.coupling_map
+        disjoint_utils.require_layout_isolated_to_component(dag, target)
```

Comment:
```suggestion
        disjoint_utils.require_layout_isolated_to_component(dag, self.coupling_map if self.target is None else self.target)
```

Just my preference, but avoiding assignment to a variable seems nice in this case since `require_layout_isolated_to_component` is the only thing that (hopefully) deals in terms of a target and coupling map union.

Reply:
Heh, I personally never like the inline syntax I find it harder to read. But sure I can update this here and in the other passes doing the same

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9927/commits/9600c17f8f4505ecc9bf3c0bb1e886b8fab27be7